liferay-commerce/liferay-portal
# Organizations Chart - Basic interactions

Hi guys, this PR is here to allow any frontender to contribute to / review the organisation. It's just the first one with the most basic interactions (3-4 will follow). Any help is welcome and highly appreciated :)

Just a small introduction:
This PR is just a draft as, to make the chart working, there are a bunch of changes to be done: the organizations/accounts API need some update, there's  some cleaning work to be done concerning the BE side and the jsp where the component will be required is still missing.

**What this PR allows:**

- [x] navigating through organisations, accounts and users in a static environment
- [ ] creating new organizations via Modal
- [ ] creating or adding accounts within organizations
- [ ] inviting users within accounts
- [ ] removing / deleting entities via dropdown menu
- [ ] update entities connection via drag & drop
- [ ] multi selection via shift key
- [ ] paths highlighting at mouseover
- [ ] actions available when user has the right permissions
- [ ] it works in portal
- [ ] it supports a table view
- [ ] it supports a map view

**How to test it?** 

I've included in the this PR some commits from @luca-pellizzon which will lately be discarded.
The quickest way is deploying all the `accounts` related and `headless/admin-user` modules. 
Then running `yarn watch` from `modules/dxp/apps/commerce/commerce-organization-web/`
Finally open `test/dev/entry.js` and update the prop `rootOrganizationId` passing the the root organisation id (here's the endpoint `http://localhost:8080/o/headless-admin-user/v1.0/organizations/`).

If you wanna check the current state of the ORG-CHART, feel free to check(out) my `ORG-CHART` branch. I didn't push everything just to give a smaller scope to review. 

**Design**

Here's the final expected result: https://www.figma.com/file/Fgh6jD5qPBr8srVZlmxAWc/(m)-commerce-classic-theme-01a-ac?node-id=5037%3A95299. 


Thanks in advance,
Fabio

cc @nhpatt, @riccardo-alberti
